src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -181,9 +181,10 @@ def _executeProducer(self, reqStatus):
         try:
             requestRecords = {}
             for status in reqStatus:
+                numRequestRecords = len(requestRecords)
                 requestRecords.update(self.getRequestRecords(status))
```

Comment:
Isn't it easier to just separate this line into two, a) fetch requests, b) update requestRecords. In between them, you print the amount of requests in that specific status. This avoids the arithmetics with dict/lists.

Reply:
I know that in the code it looks much cleaner... but then you need save the output with the enormous list of `requestRegcords` from the given new `status` in another variable and you need to to live with that variable up until the end of the whole call to`_executeProducer()`, while in the case how I made it, one should never save anything out of the scope of `update()` and since the `len()` builtin is already of complexity of O(1) you basically have the best performance with the minimum memory footprint here. So my opinion is we should bare with the little bit unreadable code here for the sake of performance.

Reply:
You have a good point. I'm curious to know whether in practice this is what happens though (in case you have profiled it). Otherwise, the code is good to go indeed.